### PR TITLE
Fix dashboard edit button is disabled

### PR DESCRIPTION
**Pull request by mistercrunch**

### CODE REVIEW

File: superset/assets/javascripts/dashboard/components/Controls.jsx [hunk 1]
```diff
@@ -81,26 +80,29 @@ class Controls extends React.PureComponent {
...
             window.location = `/dashboardmodelview/edit/${dashboard.id}`;
           }}
-          tooltip="Edit this dashboard's property"
+          tooltip="Edit this dashboard's properties"
         >
           <i className="fa fa-edit" />
         </Button>
         <SaveModal
           dashboard={dashboard}
+          disabled={!dashboard.dash_save_perm}
```
Comment (ascott) [inline superset/assets/javascripts/dashboard/components/Controls.jsx]: why does `<SaveModal />` need a `disabled` prop?

File: superset/assets/javascripts/components/ModalTrigger.jsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ const propTypes = {
   bsSize: PropTypes.string,
   className: PropTypes.string,
   tooltip: PropTypes.string,
+  disabled: PropTypes.bool,
```
Comment (ascott) [inline superset/assets/javascripts/components/ModalTrigger.jsx]: in what instance are we passing a `disabled` flag for `<ModalTrigger />`?

Comment (mistercrunch) [inline superset/assets/javascripts/components/ModalTrigger.jsx]: Oh right, I ended up backtracking on this approach when I realized I could disable my own `triggerNode` prop that I pass to it.

File: superset/assets/javascripts/dashboard/components/Controls.jsx (hunk 1 above)
Comment (mistercrunch) [inline superset/assets/javascripts/dashboard/components/Controls.jsx]: If the user does not have the permission to save the dashboard.

**Comment by ascott** [comment]
LGTM

**Pull request merged**